### ISSUE TITLE
Feature request: Combined pickers

### ISSUE THREAD

**Issue by Sarpedion** [vuetifyjs/vuetify#459]
Combining date and time pickers in a single picker would provide better user experience while, for example, scheduling an event, because an event could be scheduled in a single field, instead of two fields.

An example implementation:

![image](https://cloud.githubusercontent.com/assets/16826299/25531139/08fa0b90-2c29-11e7-8bd9-bdc1d777349d.png)

Furthermore, in time picker, disabling selection of minutes might also be a useful feature, for situations where it is not desirable for users to select minutes.

**Comment by johnleider** [comment]
This is something that would not be part of the Vuetify core, but the front-end addon pack planned later this year. For now I'll let the issue reside here and will move it over once that repo is started.

**Comment by donPuerto** [comment]
What do you mean by addon pack? Let us know your future plans?

**Comment by johnleider** [comment]
After the Vuetify core is done I'll begin work on a frontend and backend package.

**Comment by donPuerto** [comment]
very nice, so can we recommend a request feature like calendar... By the way do we have walkthrough on how to integrate this to Laravel ?

**Comment by johnleider** [comment]
No walk-through, but it would be helpful. Once everything stabilizes with the package, I have lots of developer QoL updates I'd like to make in the way of guides and tutorials.

**Comment by scgm11** [comment]
any update on this?

**Comment by AGPDev** [comment]
+1

**Comment by bdeo** [comment]
Please do not comment +1 on issues, comments are for discussion. If you'd like to show your interest in this ticket, please Vote on OP's comment with the 👍 emoji.

**Comment by maxzhou0** [comment]
the date picker and time picker are not very handy and i hope there's a new component to improve these:

- The binding data type should be Date 
- expose data during the input.  for example when user pick minute, how can i know which hour and date they just picked?  the allowed-minutes cannot not return expected value without the informations.
- customize the input order. default is date -> hour -> minute ,but date->hour or  hour->minute should be ok,

**Comment by johnleider** [comment]
In the past we used the Date object but it became a nightmare in regards to localization. Your suggestions are also unrelated to this particular feature request @maxzhou0 . Please create a new Feature Request at https://issues.vuetifyjs.com

**Comment by scgm11** [comment]
are you evaluating using moment?

**Comment by johnleider** [comment]
No. I believe the goal is to ensure we can easily hook up to other libs/functionality.

**Comment by mviens** [comment]
I also would like this combined component. I have found the Vuetify Date and Time pickers to be quite clunky compared to the other components. The best pickers I have found (Date, Time, and Date/Time) are here (https://material-ui-pickers.dev/api/datetime-picker) but these are React-based. I would like to see Vuetify match the styling, features, and smoothness with which these work.

**Comment by douglasg14b** [comment]
Was hoping vuetify had combined date-time pickers. This would go a long ways towards adding that extra bit of usability to the framework. It's awkward to have separate date and time fields.

**Comment by The-Mojoo** [comment]
I've also been in great need of this component recently, and the experience is so important that I've tried other components and it's always been indecent, and I'd hope Vuetify to support this feature. I noticed that this issue was raised 2 years ago, has not yet been resolved, and was not set to new Component in the 2.0 version, and I sincerely hope that developers will be able to value this feature.

**Comment by moonsphinx** [comment]
This is such an important component, I'm even surprised how deeply I need it at the moment. I hope it comes very soon to Vuetify.

**Comment by Mikilll94** [comment]
Any progress on this?

**Comment by cristianmatox** [comment]
I made a workaround for this in the meanwhile with this component: https://codepen.io/xristian/pen/VoLRYa

**Comment by positivethinking639** [comment]
@cristianmatox Can I custom datetimepicker to be like this https://ibb.co/NptQsgR? I get it from here : https://vuetifyjs.com/en/getting-started/consulting-and-support. is there a component like that in vuetify?

**Comment by Mikilll94** [comment]
Is it going to be implemented soon?

**Comment by douglasg14b** [comment]
@cristianmatox If you add `z-index: 1;` to the AM/PM buttons the dropshadow won't be clipped.

**Comment by andreafonso** [comment]
@johnleider can you give an ETA to this?

**Comment by makeit-mh** [comment]
Hey, 

I'm found 2 great concepts about DateTimePicker. 
First with tabs and make for Vuetify:
https://darrenfang.github.io/vuetify-datetime-picker/
And second with (in my opinion) better TimePicker concept:
https://mariomka.github.io/vue-datetime/

What do you think about that?

**Comment by sh7dm** [comment]
This'll probably be added in v3 with a complete pickers rewrite.

**Comment by abdelaziz321** [comment]
In case anybody came here for a very quick solution without using additional plugins.

```vue
<template>
  <div>
    <v-dialog ref="offerStartedDateDialog" :return-value.sync="offer_started" width="650px">
      <template v-slot:activator="{ on }">
        <v-text-field
          dense
          readonly
          full-width
          outlined
          dir="ltr"
          v-on="on"
          clearable
          label="بداية الخصم"
          prepend-icon="mdi-calendar-month"
          :value="offer_started_date"
        ></v-text-field>
      </template>
      <div class="white">
        <div class="d-flex justify-space-between">
          <div>
            <v-date-picker
              v-model="offer_started.date"
              scrollable
              width="350px"
              flat
              locale="ar"
            />
          </div>
          <div>
            <v-time-picker
              locale="ar"
              flat
              v-model="offer_started.time"
            />
          </div>
        </div>
        <div class="text-center py-4">
          <v-btn color="primary" @click="$refs.offerStartedDateDialog.save(offer_started)">حفظ</v-btn>
        </div>
      </div>
    </v-dialog>
  </div>
</template>


<script>
export default {
  name: "record-form",


  data: function () {
    return {
      offer_started: {
        time: '00:00',
        date: new Date().toISOString().substr(0, 10)
      },
      form: {},
    };
  },


  computed: {
    offer_started_date: function () {
      if (!(this.offer_started.date && this.offer_started.time)) {
        return `${new Date().format('Y-m-d')} 00:00:00`;
      }

      return `${this.offer_started.date} ${this.offer_started.time}:00`
    }
  },


  methods: {
    addPrices(form) {
      this.form.offer_started_date = this.offer_started_date;

      // ...
    }
};
</script>
```

\- and here is the result:

![Screenshot from 2020-08-27 18-24-17](https://user-images.githubusercontent.com/21992961/91471181-fb2ca680-e895-11ea-9e0f-e40fbff6db4d.png)


![Screenshot from 2020-08-27 18-30-57](https://user-images.githubusercontent.com/21992961/91471192-0253b480-e896-11ea-83a5-c88357292d33.png)

**Comment by talski** [comment]
I came across this problem, until no official solution, here's mine

https://codepen.io/talski/pen/WNoKrJL

Just customize as you like and wrap in a custom component.

**Comment by gregveres** [comment]
Like @mviens, I think of all the suggestions in this thread, this one is the best: https://material-ui-pickers.dev/api/datetime-picker one to mimic.

**Comment by douglasg14b** [comment]
@gregveres  That's a dead link for me , 404.

**Comment by jacekkarczmarczyk** [comment]
<https://material-ui-pickers.dev/api/DateTimePicker>

**Comment by gregveres** [comment]
Here is the proper link to the demo. Sorry about that. My mistake:
https://material-ui-pickers.dev/demo/datetime-picker

@jacekkarczmarczyk  posted the link to the api. I am less interested in matching their api. I think the visual of that date time picker is great.

**Comment by gregveres** [comment]
I have created a v-date-time-picker.vue prototype in Vuetify 2.0. I posted it as a [gist](https://gist.github.com/gregveres/2e24931280749fc5dca828780bedf7dc). I now want to create a feature request for v-time-picker.vue because it is inconsistent with v-date-picker. V-data-picker allows control over which picker is visible, but v-time-picker does  not. This really happers the usability of the combined picker because once the user is viewing the minutes, there is absolutely on way to get the time-picker to cycle back to the hours picker. 

anyway, if someone wants to use this prototype, they are free to. I am hoping that when we get to vuetify 3, I can replace this with a native version.

**Comment by volarname** [comment]
we need combined picker,
also we need another that clock picker for time

**Comment by PHKenny** [comment]
Our company created a datetime picker using native components of Vuetify 2.6, check it out https://gitlab.com/goldenm-software/open-source-libraries/vuetify-datetime-picker

**Comment by J-Sek** [comment]
I wish somebody could moderate (remove) links to "material-ui-pickers.dev" that redirect to "wheelclothing.cz". @gregveres , @jacekkarczmarczyk

**Comment by J-Sek** [comment]
It looks very much like a specific design that is not very common nowadays. I am personally leaning towards something [like this](https://play.vuetifyjs.com/playgrounds/-mI3Tg)

<details>
<summary>Preview</summary>

<img width="1429" height="1261" alt="Image" src="https://github.com/user-attachments/assets/05091145-2578-4a0c-943b-c747daf6441b" />

</details>

**Comment by AndreyYolkin** [comment]
> It looks very much like a specific design that is not very common nowadays. I am personally leaning towards something [like this](https://play.vuetifyjs.com/playgrounds/-mI3Tg)
> 
> Preview

+1, We have pretty similar design, even with hours and minutes separated
